enhancements/installer/log-config.md

```diff
@@ -0,0 +1,189 @@
+and we need to decorate output that would occur if there are
+errors in the install config and it cannot be loaded.
+
+The log-config.yaml file would be loaded if present in the install directory,
+as well as adding a command line flag  --log-config  to accept a path to the config file.
+
+Users should be able to provide their config in a file that looks, for example, like:
+
+```yaml
+fields:
+  key1: value1
+  key2: value2
+  key3: value3
+level: debug
+```
+
+### Workflow Description
+
+1. Installer user specifies configuration in `log-config.yaml`
+2. `log-config.yaml` is placed in install dir or path is specified with a flag
+3. Installer uses `log-config.yaml` for any Installer commands
+
+
+### API Extensions
+
+n/a
+
+### Risks and Mitigations
+
+Adding the ability to configure logging comes with a risk that the
+logging could have errors or misconfiguration. There is an Open
+Question about whether logging errors should fail the install or
+just log a warning message:
+
+If users include a `log-config.yaml` file, they intend to configure
+logging. If there is an error in that file, should we prevent installs
+until that configuration is properly handled? Or should we make the
+best effort to complete the install although the logging is not
+properly configured? 
+
+My weakly held opinion is that we should prevent attempting installs if
+`log-config.yaml` is supplied but is misconfigured.
+
+### Drawbacks
+
+This enhancement proposes a new way to introduce logging configuration.
+The existing methods we have are: install config, CLI flags, and 
+environment variables (not supported for customer user). By introducing
+another method of configuration, we risk confusion. I do not think
+this is a serious drawback or perhaps even a valid one.
+
+## Design Details
+
+### Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ 1. Should logging misconfiguration prevent installs? (see Risks & Mitigations)
+ 2. Should log-config.yaml be versioned according to k8s API versioning?
+
+### Test Plan
+
+For e2e-tests, it would be possible to drop a log-config.yaml
+file in the configuration stags and ensure that logs are
+properly configured.
+
+### Graduation Criteria
+
+The intended audience for this feature is ARO, so it would be possible
+to have this used by our managed services providers before releasing
+to customers, but this level of precaution may not be needed for this 
+feature.
+
+### Upgrade / Downgrade Strategy
+
+n/a
+
+### Version Skew Strategy
+
+n/a 
+
+### Operational Aspects of API Extensions
+
+n/a
+
+#### Failure Modes
+
+As discussed above, it is an open question about whether logging
+misconfiguration should prevent an install from launching.
+
+## Alternatives
+
+* install config has been ruled out for reasons discussed above
+* env vars are not supported for customer use
```

Comment:
Do we actually have a use case for this for customers? It seems like mostly a managed services thing.

Reply:
No. If we think there is limited usefulness to this feature we could consider keeping it "internal only" for now. I will try to work through the Graduation Criteria, if I'm using/understanding that correctly, in my next edit.